WalletWasabi/Services/UpdateManager.cs

```diff
@@ -0,0 +1,253 @@
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using WalletWasabi.Logging;
+using WalletWasabi.Microservices;
+using WalletWasabi.Models;
+using WalletWasabi.Tor.Http;
+
+namespace WalletWasabi.Services;
+
+public class UpdateManager
+{
+	private string InstallerName { get; set; } = "";
+
+	public UpdateManager(string dataDir, IHttpClient httpClient)
+	{
+		DataDir = dataDir;
+		DownloadsDir = Path.Combine(DataDir, "Downloads");
+		HttpClient = httpClient;
+	}
+
+	public async void UpdateStatusChangedAsync(UpdateStatus updateStatus)
+	{
+		bool updateAvailable = !updateStatus.ClientUpToDate || !updateStatus.BackendCompatible;
+		Version targetVersion = updateStatus.ClientVersion;
+		Exception? downloadException = null;
+		if (!updateAvailable)
+		{
+			return;
+		}
+
+		byte retries = 0;
+		Logger.LogInfo($"Trying to download new version: {targetVersion}");
+		do
+		{
+			try
+			{
+				bool isReadyToInstall = await GetInstallerAsync(targetVersion).ConfigureAwait(false);
+				Logger.LogInfo($"Version {targetVersion} downloaded successfuly.");
+				updateStatus.IsReadyToInstall = isReadyToInstall;
+				break;
+			}
+			catch (Exception ex)
+			{
+				downloadException = ex;
+				Logger.LogWarning($"Geting version {targetVersion} failed. Retrying...");
+			}
+		} while (retries++ < 3);
+
+		if (retries == 2 && downloadException is { })
+		{
+			Logger.LogError($"Geting version {targetVersion} failed with error.", downloadException);
+		}
```

Comment:
Extra logic and variable just to log differently? IMHO feels a bit redundant.

We don't even log out `downloadException`.